argus/unit_tests/action_manager/test_windows.py

```diff
@@ -13,64 +13,818 @@
+
+    def test_get_installation_script_successful(self):
+        self._test_get_installation_script()
+
+    def test_get_installation_script_exception(self):
+        self._test_get_installation_script(exc=exceptions.ArgusTimeoutError)
+
+    @mock.patch('argus.action_manager.windows.WindowsActionManager'
+                '.wait_boot_completion')
+    @mock.patch('argus.action_manager.windows.WindowsActionManager'
+                '.download_resource')
+    def _test_sysprep(self, mock_download_resource, mock_wait_boot_completion,
+                      download_exc=None, run_exc=None, wait_exc=None):
+        cmd = r"C:\{}".format(
+            test_utils.SYSPREP_RESOURCE_LOCATION.split('/')[-1])
+
+        self._client.run_remote_cmd = mock.Mock()
+
+        if download_exc:
+            mock_download_resource.side_effect = download_exc
+            with self.assertRaises(download_exc):
+                self._action_manager.sysprep()
+            self._client.run_remote_cmd.assert_not_called()
+            return
+
+        if wait_exc:
+            mock_wait_boot_completion.side_effect = wait_exc
+            with self.assertRaises(wait_exc):
+                self._action_manager.sysprep()
+            return
+
+        if run_exc:
+            self._client.run_remote_cmd.side_effect = run_exc
+            with test_utils.LogSnatcher('argus.action_manager.windows.Windows'
+                                        'ActionManager.sysprep') as snatcher:
+                self.assertIsNone(self._action_manager.sysprep())
+            self.assertEqual(snatcher.output[-2:],
+                             ['Currently rebooting...',
+                              'Wait for the machine to finish rebooting ...'])
+        else:
+            self.assertIsNone(self._action_manager.sysprep())
+
+        mock_download_resource.assert_called_once_with(
+            test_utils.SYSPREP_RESOURCE_LOCATION, cmd)
+        mock_wait_boot_completion.assert_called_once_with()
+
+    def test_sysprep_successful(self):
+        self._test_sysprep()
+
+    def test_sysprep_download_resource_fail(self):
+        self._test_sysprep(download_exc=exceptions.ArgusTimeoutError)
+
+    def test_sysprep_wait_boot_completion_fail(self):
+        self._test_sysprep(wait_exc=exceptions.ArgusTimeoutError)
+
+    def test_sysprep_run_remote_cmd_exc(self):
+        self._test_sysprep(run_exc=requests.Timeout)
+
+    @mock.patch('argus.action_manager.windows.WindowsActionManager'
+                '.exists')
     def test_git_clone_exists(self, mock_exists):
         mock_exists.return_value = True
-        self.assertRaises(exceptions.ArgusCLIError,
-                          self.action_manager.git_clone,
-                          'fake_url', 'fake_location')
 
-    @mock.patch('argus.action_manager.windows.WindowsActionManager.'
-                'exists')
+        with self.assertRaises(exceptions.ArgusCLIError):
```

Comment:
Please undo these changes, this commit has the purpose to add unit tests, not to rewrite already existing ones.